src/plugins/list.js

```diff
@@ -11,7 +13,35 @@ function List() {
    * @param {Editor} editor
    * @param {Function} next
    */
-  const onEnter = (event, editor, next) => next();
+  const onEnter = (event, editor, next) => {
+    const { value } = editor;
+    const { startBlock } = value;
+    event.preventDefault();
+
+    // Hitting enter on a blank list item will break out of the enclosing list
+    if (isSelectionList(value) && startBlock.text.length === 0) {
+      editor.withoutNormalizing(() => {
+        event.preventDefault();
+        editor
+          .setBlocks(CONST.PARAGRAPH)
+          .unwrapBlock(CONST.LIST_ITEM)
+          .unwrapBlock(CONST.OL_LIST)
+          .unwrapBlock(CONST.UL_LIST);
+      });
+      return false;
+    }
+
+    // Hitting enter on a non-empty list item will add a new list_item
+    if (isSelectionList(value) && startBlock.text.length !== 0) {
+      editor.withoutNormalizing(() => {
+        event.preventDefault();
+        editor.splitBlock().unwrapBlock(CONST.LIST_ITEM).wrapBlock(CONST.LIST_ITEM);
+      });
+      return true;
```

Comment:
What is the effect of this? Shouldn't it return `next()` instead?

Reply:
I think `next()` is resulting in there being a `paragraph` in the `list_item` and a new `paragraph` below, as if a sibling of the `list_item`.

Reply:
I'm confused about this as well. What should be the type returned by `onEnter` ?

Reply:
I'm unsure if it should be returning a type. @dselman you may have some insight?

Reply:
What is `.unwrapBlock(CONST.LIST_ITEM).wrapBlock(CONST.LIST_ITEM)` doing on line 37? Why would we unwrap a list item just to wrap it again?